tests/framework/CodeInterfaceTests/RAVEN/Code/projectile.py

```diff
@@ -0,0 +1,174 @@
+  """
+  return x0 + v*t
+
+def y_pos(y0,v,t):
+  """
+    Calculates the analytic vertical position in time
+    @ In, y0, float, initial vertical position
+    @ In, v, float, velocity of the projectile
+    @ In, t, float, time of flight
+    @ Out, y_pos, float, vertical position
+  """
+  return y0 + v*t - 4.9*t*t
+
+def calc_vel(y0, y, v0, ang, g=9.8):
+  E_m = 0.5 * v0*v0 + g*y0
+  vel = np.sqrt(v0*v0 - 2*g*(y-y0))
+  x_vel = v0 * np.cos(ang)
+  y_vel = np.sqrt(vel*vel - x_vel*x_vel)
+  return x_vel, y_vel, vel
+
+def current_angle(v0, ang, vel):
+  return np.arccos(v0 * np.cos(ang) / vel)
+
+def run(raven, inputs):
+  vars = {'x0': get_from_raven('x0', raven, 0),
+          'y0': get_from_raven('y0', raven, 0),
+          'v0': get_from_raven('v0', raven, 1),
+          'angle': get_from_raven('angle', raven, 45),
+          'timeOption': get_from_raven('timeOption', raven, 0)}
+  res = main(vars)
+  raven.x = res['x']
+  raven.y = res['y']
+  raven.t = res['t']
+  raven.r = res['r'] * np.ones(len(raven.x))
+  raven.v = res['v']
+  raven.a = res['a']
+
+def get_from_raven(attr, raven, default=None):
+  return np.squeeze(getattr(raven, attr, default))
+
+def main(Input):
+  x0 = Input.get('x0', 0)
+  y0 = Input.get('y0', 0)
+  v0 = Input.get('v0', 1)
+  ang = Input.get('angle', 45)
+  g = Input.get('g', 9.8)
+  timeOption = Input.get('timeOption', 0)
+  ang = ang * np.pi / 180
+  if timeOption == 0:
+    ts = np.linspace(0,1,10)
+  else:
+    # due to numpy library update, the return shape of np.linspace
+    # is changed when an array-like input is provided, i.e. return from time_to_ground
+    ts = np.linspace(0,time_to_ground(v0,ang,y0),10)
+
+  vx0 = np.cos(ang)*v0
+  vy0 = np.sin(ang)*v0
+  r = prange(v0,ang,y0)
+
+  x = np.zeros(len(ts))
+  y = np.zeros(len(ts))
+  v = np.zeros(len(ts))
+  a = np.zeros(len(ts))
+  for i,t in enumerate(ts):
+    x[i] = x_pos(x0,vx0,t)
+    y[i] = y_pos(y0,vy0,t)
+    vx, vy, vm = calc_vel(y0, y[i], v0, ang, g)
+    v[i] = vm
+    a[i] = current_angle(v0, ang, vm)
+  t = ts
+  res = {'x': x, 'y': y, 'r': r, 't': ts, 'v': v, 'a': a,
+         'x0': x0, 'y0': y0, 'v0': v0, 'ang': ang, 'timeOption': timeOption}
+  return res
+
+#can be used as a code as well
+if __name__=="__main__":
+  import sys
+  textOutput = False
+  if '-i' not in sys.argv:
+    raise IOError("INPUT MUST BE PROVIDED WITH THE -i nomenclature")
+  if '-o' not in sys.argv:
+    raise IOError("OUTPUT MUST BE PROVIDED WITH THE -o nomenclature")
+  if '-text' in sys.argv:
+    textOutput = True
+  inFile = sys.argv[sys.argv.index('-i')+1]
+  outFile = sys.argv[sys.argv.index('-o')+1]
+  #construct the input
+  Input = {}
+  for line in open(inFile,'r'):
+    arg, val = (a.strip() for a in line.split('='))
+    if arg == 'aux':
+      auxFile = val
+    else:
+      Input[arg] = float(val)
+  # check auxfile exists
+  if not os.path.isfile(auxFile):
+    raise IOError('Aux file not found: {}'.format(auxFile))
```

Comment:
Checking the sys.argv '-i -o' and the existence of aux file, only needed for this test